README.md

```diff
@@ -3,11 +3,150 @@
+
+This project doesn't really contain Go project, but some of the code generation tools (`protoc-gen-openapi`, for
+example) are written in Go and installed using `go install ...`. So you will need the Go compiler and tools. If your
+operating system provides a recent enough version, at least 1.22.9, then you can use it directly. If not you can
+download and install it manually. For example, you can install it to `~/go` with something like this:
+
+    $ wget https://go.dev/dl/go1.22.9.linux-amd64.tar.gz
+    $ tar xvf go1.22.9.linux-amd64.tar.gz
+
+This is just an example, make sure to download the version appropriate for your operating system and architecture.
+
+Later you will configure your environment (via the `.envrc` file ) to use that version.
+
+### Install the `direnv` tool
+
+Follow the instructions in the [`direnv`](https://direnv.net) page to install and enable it. This will be used to create
+an environment for the project without interfering with other settings that you may need for your system or for other
+projects.
+
+### Prepare the project directory
+
+Create a directory where you will have the files for the project, for example `~/fulfillment-api`. Note that this is
+not where you will check-out the source code of the project: that will go into a `repository` sub-directory. But before
+that create a `.envrc` file in that directory similar to this:
+
+```shell
+# Configure Python:
+export VIRTUAL_ENV=".venv"
+layout python
+
+# Configure Go:
+export GOROOT="${HOME}/go"
+export GOPATH="${PWD}/.local"
+export GOBIN="${PWD}/.local/bin"
+PATH_add "${GOROOT}/bin"
+PATH_add "${PWD}/.local/bin"
+```
+
+Make sure to adjust these settings for your environment, in particular the `GOROOT` should point to the directory
+where you installed Go.
+
+The next time you go into the project directory, the `direnv` tool try to automatically load those settings, but if it
+is the first time it will do nothing and ask you to give it permission. You will see an error message like this:
+
+   direnv: error /home/builder/fulfillment-api/.envrc is blocked. Run `direnv allow` to approve its content
+
+Run that `direnv allow` command:
+
+    $ cd ~/fulfillment-api
+    direnv: error /home/builder/fulfillment-api/.envrc is blocked. Run `direnv allow` to approve its content
+
+    $ direnv allow
+    direnv: loading ~/fulfillment-api/.envrc
+    direnv: export +GOBIN +GOPATH +GOROOT +PGDATABASE +PGHOST +PGPASSFILE +PGUSER +VIRTUAL_ENV ~PATH
+
+Any time you enter that directory now you will have the right environment variables, and when you leave the directory it
+will clean them up.
+
+In addition, the first time it will automatically create a new Python virtual environment inside the `.venv` directory,
+and will activate it when you enter the directory.
+
+Verify that the Python and Go versions are set correctly:
+
+    $ python --version
+    Python 3.12.5
+
+    $ go version
+    go version go1.22.9 linux/amd64
+
+### Clone the git repository
```

Comment:
shouldn't this section belong to the fulfilment-service repo?

Reply:
Ah, yes. I copied it from https://github.com/jhernand/fulfillment-service and forgot to replace `-service` with `-api`, I am fixing it, thanks @adriengentil.

Reply:
Done, please check again.